File: frontend/src/types/report.ts
/** Mirrors backend/app/schemas/report.py. */

export interface StatusCount {
  status: string;
  count: number;
}

export interface JobApplicationCount {
  job_id: string;
  job_title: string;
  count: number;
}

export interface ScreeningSummary {
  total_runs: number;
  completed: number;
  failed: number;
  average_score: number | null;
}

export interface AssessmentSummaryStats {
  total_invitations: number;
  submitted: number;
  passed: number;
}

export interface CampusDriveCount {
  drive_id: string;
  drive_name: string;
  application_count: number;
}

export interface ReportOverview {
  total_jobs: number;
  open_jobs: number;
  total_candidates: number;
  total_applications: number;
  applications_by_status: StatusCount[];
  applications_by_job: JobApplicationCount[];
  screening: ScreeningSummary;
  assessments: AssessmentSummaryStats;
  campus_drives: CampusDriveCount[];
}

File: backend/app/schemas/report.py
import uuid

from pydantic import BaseModel


class StatusCount(BaseModel):
    status: str
    count: int


class JobApplicationCount(BaseModel):
    job_id: uuid.UUID
    job_title: str
    count: int


class ScreeningSummary(BaseModel):
    total_runs: int
    completed: int
    failed: int
    average_score: float | None


class AssessmentSummaryStats(BaseModel):
    total_invitations: int
    submitted: int
    passed: int


class CampusDriveCount(BaseModel):
    drive_id: uuid.UUID
    drive_name: str
    application_count: int


class ReportOverview(BaseModel):
    """Every field here is computed from persisted data at request time
    (CLAUDE.md § 2: "Reports != hardcoded numbers"), tenant-scoped like
    every other recruiter-facing query."""

    total_jobs: int
    open_jobs: int
    total_candidates: int
    total_applications: int
    # Candidate-outcome metrics for the Overview dashboard (SIGVITAS
    # platform overhaul § 15) — each a direct count of
    # `ApplicationStatus.{SELECTED,REJECTED,HIRED}`, tenant-scoped and
    # excluding soft-deleted applications like every other field here.
    selected_candidates: int
    rejected_candidates: int
    hired_candidates: int
    applications_by_status: list[StatusCount]
    applications_by_job: list[JobApplicationCount]
    screening: ScreeningSummary
    assessments: AssessmentSummaryStats
    campus_drives: list[CampusDriveCount]
